core/control-plane/control-plane-api/src/main/java/org/eclipse/dataspaceconnector/controlplane/api/ControlPlaneApiExtension.java

```diff
@@ -0,0 +1,125 @@
+ */
+@Provides(ControlPlaneApiUrl.class)
+@Extension(value = ControlPlaneApiExtension.NAME)
+public class ControlPlaneApiExtension implements ServiceExtension {
+
+    public static final String NAME = "Control Plane API";
+
+    public static final String CONTROL_PLANE_API_CONFIG = "web.http.controlplane";
+    public static final String CONTROL_PLANE_CONTEXT_ALIAS = "controlplane";
+
+    public static final int DEFAULT_CONTROL_PLANE_API_PORT = 8384;
+    public static final String DEFAULT_CONTROL_PLANE_API_CONTEXT_PATH = "/api/v1/controlplane";
+
+    private int port = DEFAULT_CONTROL_PLANE_API_PORT;
+    private String path = DEFAULT_CONTROL_PLANE_API_CONTEXT_PATH;
+
+    @Inject
+    private WebServer webServer;
+    @Inject
+    private WebService webService;
+    @Inject
+    private AuthenticationService authService;
+    @Inject
+    private TransferProcessManager transferProcessManager;
+    @Inject
+    private Hostname hostname;
+
+    @Override
+    public String name() {
+        return NAME;
+    }
+
+    @Override
+    public void initialize(ServiceExtensionContext context) {
+        var monitor = context.getMonitor();
+        var alias = CONTROL_PLANE_CONTEXT_ALIAS;
+
+        var config = context.getConfig(CONTROL_PLANE_API_CONFIG);
+        if (config.getEntries().isEmpty()) {
+            monitor.warning(format("Settings for [%s] and/or [%s] were not provided. Using default" +
+                    " value(s) instead.", CONTROL_PLANE_API_CONFIG + ".path", CONTROL_PLANE_API_CONFIG + ".path"));
+            webServer.addPortMapping(alias, port, path);
+        } else {
+            path = config.getString("path", path);
+            port = config.getInteger("port", port);
+        }
+
+        context.getTypeManager().registerTypes(TransferProcessFailStateDto.class);
+
+        monitor.info(format("Control Plane API will be available at [path=%s], [port=%s].", path, port));
+
+        webService.registerResource(alias, new TransferProcessControlApiController(transferProcessManager));
+        webService.registerResource(alias, new AuthenticationRequestFilter(authService));
+    }
+
+
+    @Provider
+    public ControlPlaneApiUrl controlPlaneApiUrl(ServiceExtensionContext context) {
+        String s = getApiUrl();
+        try {
+            var url = new URL(s);
+            return () -> url;
+        } catch (MalformedURLException e) {
+            context.getMonitor().severe("Error creating callback endpoint", e);
+            throw new EdcException(e);
+        }
+    }
+
+    @NotNull
+    private String getApiUrl() {
+        var s = hostname.get();
+
+        if (!s.startsWith("http")) { // a hostname should never have a protocol prefix, but just to be safe
+            s = "http://" + s;
+        }
+        if (!s.matches(".*:([0-9]){1,5}$")) { // a hostname also shouldn't have a port, but again, to be sure
+            s += ":" + port;
+        }
```

Comment:
Shouldnt these checks be performed in the `Hostname` class?

Reply:
I agree, in `CoreServicesExtension` we should ensure that this is only an hostname (so no protocol nor port added)

Reply:
I got it, but these checks could be moved to one single point, and validate a config setting seems like a sane approach instead of introducing a whole lot of logic aroung

Reply:
So we can just assume here that the Hostname is clean, got it :)